commands/publish_dafsa.py

```
@@ -0,0 +1,124 @@
+from requests.exceptions import HTTPError
+
+import os
+import subprocess
+from kinto_http import Client, KintoException
+
+import json
+
+
+COMMIT_HASH_URL = (
+    "https://api.github.com/repos/publicsuffix/list/commits?path=public_suffix_list.dat"
+)
+LIST_URL = (
+    "https://raw.githubusercontent.com/publicsuffix/list/master/public_suffix_list.dat"
+)
+
+MAKE_DAFSA_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/make_dafsa.py"
+PREPARE_TLDS_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/prepare_tlds.py"
+
+SERVER = "https://kinto.dev.mozaws.net/v1"
+AUTH = (os.getenv("USERNAME"), os.getenv("PASSWORD"))  # (username, password)
+
+BUCKET_ID = "firefox-core-network-dns"
+COLLECTION_ID = "public-suffix-list"
+RECORD_ID = "latest-commit-hash"
+
+
+def handle_request_errors(func):
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash():
+    response = requests.get(COMMIT_HASH_URL)
+    response.raise_for_status()
+    return response.json()[0]["sha"]
+
+
+@handle_request_errors
+def download_resources(directory, *urls):
+    for url in urls:
+        # file_location is found by appending the file_name(at the end of url string) to temp directory
+        file_name = os.path.basename(url)
+        file_location = os.path.join(directory, file_name)
+        print(file_location)
+        response = requests.get(url, stream=True)
+        response.raise_for_status()
+
+        with open(file_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+                else:
+                    print("Error!!!")
```

What error is this? We should raise an exception if something bad happened